### PR TITLE
feat(modals): add tag prop to Modal, Drawer, and Tooltip Modal components

### PR DESCRIPTION
**Pull request by jgorfine-zendesk**
## Description

Following the precedence set in other components (e.g., Ellipsis), this PR adds the `tag` prop to the Modal Header component. This will enable consumers to customize the underlying HTML element, with the intention being that they pass in a heading (h1-h2) tag. This would help us meet WCAG 2.1 [Success Criterion 1.3.1 Info and Relationships](https://www.w3.org/TR/WCAG21/#info-and-relationships) (level A), as well as [Success Criterion 2.4.2 Page Titled](https://www.w3.org/TR/WCAG21/#page-titled) (level A).

~Separate PRs for the Drawer and Tooltip Modal to follow.~

This PR covers the Modal, Drawer, and Tooltip Modal.

### Modal

Screenshot of Chrome DevTools demonstrating the Modal component with an h2 title element

### Drawer

Screenshot of Chrome DevTools demonstrating the Drawer component with an h2 title element

### Tooltip Modal

Screenshot of Chrome DevTools demonstrating the Tooltip Modal component with an h2 title element

## Checklist

- [x] :ok_hand: design updates will be Garden Designer approved (add the designer as a reviewer)
- [x] :globe_with_meridians: demo is up-to-date (`yarn start`)
- [x] :arrow_left: renders as expected with reversed (RTL) direction
- [ ] :metal: renders as expected with Bedrock CSS (`?bedrock`)
- [x] :guardsman: includes new unit tests. Maintain [existing coverage](https://coveralls.io/github/zendeskgarden/react-components?branch=main) (always >= 96%)
- [x] :wheelchair: tested for [WCAG 2.1 AA](https://www.w3.org/WAI/WCAG21/quickref/?currentsidebar=%23col_customize&levels=aaa) accessibility compliance
- [x] :memo: tested in Chrome, Firefox, Safari, and Edge

### CODE REVIEW

**Review by jzempel — CHANGES_REQUESTED** [review]
This is looking good @jgorfine-zendesk. re:

> Separate PRs for the Drawer and Tooltip Modal to follow.

Please also wrap `DrawerModal` and `TooltipModal` into this PR as a single unit of work. :pray:

**Comment by jgorfine-zendesk** [comment]
@jzempel can do!

**Comment by jgorfine-zendesk** [comment]
Hey @jzempel, just pushed the code for the Tooltip Modal and Drawer. Do we also need to include the Sheet in this work?

File: packages/modals/src/types/index.ts [hunk 1]
```diff
@@ -101,3 +112,10 @@ export interface ITooltipModalProps
    */
   zIndex?: number;
 }
+
+export interface ITooltipModalTitleProps extends HTMLAttributes<HTMLDivElement> {
+  /**
+   * Updates the element's HTML tag
+   */
+  tag?: any;
```
Comment (jgorfine-zendesk) [inline packages/modals/src/types/index.ts:120]: Do we have a mechanism for sharing types across components? If so, I'm wondering if we want to consider centralizing the `tag` interface and exporting/extending it to these components, the typography components, and any other components that may want to use this in the future.

Comment (jzempel) [reply packages/modals/src/types/index.ts:120]: There is (see various forms under https://github.com/zendeskgarden/react-components/blob/main/packages/forms/src/types/index.ts), but not across packages. And when dealing with primitives, often the TypeScript juice isn't worth the squeeze.

**Review by jzempel — APPROVED** [review]

**Comment by jzempel** [comment]
> Do we also need to include the Sheet in this work?

Let's hold off on `Sheet` and limit this PR to the modals package.

**Pull request merged**